infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -198,10 +206,17 @@ private void initializationTask() {
                 }
 
                 // Recover if flag is false
+                int recoveryRetry = RECOVERY_RETRIES;
                 while (!recovered) {
                     recovered = runRecoveryReconfiguration();
                     if (!recovered) {
-                        log.error("detectorTaskScheduler: Recovery failed. Retrying.");
+                        if (--recoveryRetry <= 0) {
```

Comment:
Use a for loop or an intstream, this syntax isn't easy to read and leaks the scope of recoveryRetry

Reply:
Fixed.